### DISCUSSION TITLE
foundry-merkle-airdrop error with openzepplin contracts /

### DISCUSSION THREAD

**Discussion by DCVglobalnetwork** [Cyfrin/foundry-full-course-cu#2439] [General]
Dear 

As I completed my foundry-merkle-airdrop course and I pushed to my github repo, I stumbled on an error . During the course, I had an error with two OpenZepplin contracts 
`ShortString.sol` and `StorageSlot.sol` so I did modify them and they were all passing the test and all was completed well 

It seems the error might stem from the use of the tload and tstore functions, which are designed to work with "transient storage," a new concept in Solidity. However, Solidity currently doesn't have built-in support for transient storage, and OpenZeppelin's standard libraries don't include tload and tstore functionalities.

Replace or comment out the tload and tstore functions and use standard storage access methods. You can modify the library to use typical storage reads and writes instead of these non-standard transient functions.

How can I solve this? Do I have to change the OpenZeppelin version of contracts?  the official OpenZeppelin StorageSlot library from a stable version, such as v4.x? 

here is my repo

[Foundry Merkle Airdrop Project](https://github.com/DCVglobalnetwork/foundry-merkle-airdrop-project)

Appreciate suggestions and guidance.

**Comment by EngrPips** [comment]
Hello @DCVglobalnetwork, I thought you mentioned that you resolved the issue by modifying the libraries.

**Reply by DCVglobalnetwork** [reply]
thank you for getting back
Well yes, I did, but when I pushed into Github, I received an error and was asked for a test again, which I did, and also the error says where these functions are missing from OpenZeppelin contracts, and I do not know what to do with. Should I change the version instead of the latest one?

**Reply by EngrPips** [reply]
Yeah, try an older version, and let's see if that helps prevent the issue you are running into.

**Reply by DCVglobalnetwork** [reply]
How should I do that?  I install the older version for all contracts or only for these two. 
These two are:
`StorageSlot.sol` and `ShortString.sol` 
By any chance, do you have any suggestions why I did not get an error while testing and deploying? All went well once pushed to Github, but it was an error.

Thank you

**Reply by DCVglobalnetwork** [reply]
```sh
forge build --sizes
```

![image](https://github.com/user-attachments/assets/85430616-b94e-43dc-b15e-b26241e6c26a)

**Reply by EngrPips** [reply]
I am unsure why you are running into problems when you push to Git Hub. I trust someone who knows will see this and help as soon as possible.
